src/dredd.coffee

```diff
@@ -41,6 +42,24 @@ class Dredd
     @transactions = []
     @runner = new Runner(@configuration)
     configureReporters(@configuration, @stats, @tests, @runner)
+    @logProxySettings()
+
+  logProxySettings: ->
+    proxySettings = []
+    for envVariable in Object.keys(process.env)
+      continue unless envVariable.toUpperCase() in PROXY_ENV_VARIABLES
+      continue unless process.env[envVariable] isnt ''
+      proxySettings.push("#{envVariable}=#{process.env[envVariable]}")
+
+    if @configuration.options.proxy
```

Comment:
Switch upper if to `if proxySettings.length` would make more sense and could save some rows, but YMMV.